universal-application-tool-0.0.1/app/views/questiontypes/RepeaterQuestionRenderer.java

```diff
@@ -36,43 +40,68 @@ public RepeaterQuestionRenderer(ApplicantQuestion question) {
+    RepeaterQuestion enumeratorQuestion = question.createEnumeratorQuestion();
+    ImmutableList<String> entityNames = enumeratorQuestion.getEntityNames();
+
+    ContainerTag enumeratorFields = div().withId(ENUMERATOR_FIELDS_ID);
+    int index;
+    for (index = 0; index < entityNames.size(); index++) {
+      enumeratorFields.with(existingEnumeratorField(Optional.of(entityNames.get(index)), index));
+    }
 
     return div()
         .withClasses(Styles.MX_AUTO, Styles.W_MAX)
         .with(
             div()
                 .withClasses(ReferenceClasses.APPLICANT_QUESTION_TEXT)
                 .withText(question.getQuestionText()),
+            div().withId(PLACEHOLDER_ID).withClass(Styles.HIDDEN).withText(PLACEHOLDER),
             div()
                 .withClasses(
                     ReferenceClasses.APPLICANT_QUESTION_HELP_TEXT,
                     Styles.TEXT_BASE,
                     Styles.FONT_THIN,
                     Styles.MB_2)
                 .withText(question.getQuestionHelpText()),
-            repeaterFields,
-            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addRepeaterEntity")))
-        .withType("button");
+            enumeratorFields,
+            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addEnumeratorEntity"))
+                .withType("button"),
+            fieldErrors(enumeratorQuestion.getQuestionErrors()));
   }
 
-  public static Tag repeaterField(
-      String placeholderText, Optional<String> existingOption, int index) {
+  /**
+   * Crete an enumerator field for existing entries. These come with a checkbox to delete during
+   * form submission.
+   */
+  private Tag existingEnumeratorField(Optional<String> existingOption, int index) {
     ContainerTag optionInput =
         FieldWithLabel.input()
-            .setFieldName(RepeaterQuestionDefinition.REPEATED_ENTITY_NAME_KEY + "[]")
-            .setPlaceholderText(placeholderText)
+            .setFieldName(question.getContextualizedPath().toString())
             .setValue(existingOption)
             .getContainer()
             .withClasses(Styles.FLEX, Styles.ML_2);
     Tag removeOptionBox =
         FieldWithLabel.checkbox()
-            .setFieldName(RepeaterQuestionDefinition.REPEATED_ENTITY_NAME_KEY + "_delete[]")
+            .setFieldName(Path.empty().join(Scalar.DELETE_ENTITY).asArrayElement().toString())
             .setValue(String.valueOf(index))
             .getContainer();
 
-    return div().withClasses(REPEATER_FIELD_CLASSES).with(optionInput, removeOptionBox);
+    return div().withClasses(ENUMERATOR_FIELD_CLASSES).with(optionInput, removeOptionBox);
+  }
+
+  /**
+   * Create an enumerator field template for new entries. These come with a button to delete itself.
+   */
+  public static Tag newEnumeratorFieldTemplate(Path contextualizedPath) {
+    ContainerTag optionInput =
+        FieldWithLabel.input()
+            .setFieldName(contextualizedPath.toString())
```

Comment:
question: do we still want the placeholder here?

Reply:
No, the placeholder gets set in the `main.ts` method. See the other comment for why I did it this way.

Super happy to change it if there's a more straightforward way to do this.

Reply:
Fixed!!! This is much more straightforward now.